src/generic_rosenbrock.jl

```diff
@@ -0,0 +1,59 @@
+struct RosenbrockTableau
+    a::Array{Real,2}
+    C::Array{Real,2}
+    b::Array{Real,1}
+    gamma::Real
+    d::Array{Real,1}
+    c::Array{Real,1}
+end
+
+macro _genConstantCache(inds,name,T)
+    quote
+        for ind in $inds
+            push!(cacheExpr.args,:($(Symbol($name,"$(ind[1])$(ind[2])"))=convert($T,$(tab.$name[ind]))))
+        end
+    end
+end
+
+function genConstantCache(tab::RosenbrockTableau,cacheName::Symbol)
+    ainds=findall(!iszero,tab.a)
+    Cinds=findall(!iszero,tab.C)
+    cacheExpr=quote end
+    @_genConstantCache ainds a :T
+    @_genConstantCache Cinds C :T
+    @_genConstantCache eachindex(tab.b) b :T
+
+end
+
+#=
+    for aind in ainds
+        push!(cacheExpr.args,:($(Symbol("a$(aind[1])$(aind[2])"))=convert(T,$(tab.a[aind]))))
+    end
+    for Cind in cinds
+        push!(cacheExpr.args,:($(Symbol("C$(cind[1])$(cind[2])"))=convert(T,$(tab.C[Cind]))))
+    end
+=#
+#---------------Constant Cache--------------
+macro RosenbrockW6S4OSConstantCache
+
+end
+
+function RosenbrockW6S4OSTableau()
+    a=[0                  0                  0                  0                  0;
```

Comment:
This should probably be a static array so the length can be used at compile time. Probably an optimization for later.

Reply:
Emmm, I thought functions called inside the macros are evaluated at the compile time.

Reply:
They are. So I guess gen_constant_cache will be ran from a macro?

Reply:
yes